Source/i_system.c

```diff
@@ -219,21 +219,22 @@ void I_InitKeyboard(void)
       SDL_SetModState(KMOD_NONE);
 }
 
-extern boolean nomusicparm, nosfxparm;
-
-void I_Init(void)
+void I_InitTimer(void)
 {
    int p;
 
+   int time_ms = SDL_GetTicks();
+   int old_time = time_ms * clock_rate / 100;
+   int delta = (time_ms * realtic_clock_rate / 100) - old_time;
+
+   basetime += delta;
```

Comment:
@fabiangreffrath I'm struggling a little here. I think we need to change the `basetime` to compensate for the speed change, but that doesn't work right.

Reply:
Hm, I am not sure if we even need gradual increase/decrease of game speed. In Crispy (HEAD), for example, I have a key to merely switch between playdemo/timedemo behaviour, and this comes with a lot less hassle, obviously.

Reply:
We have this cool game speed feature, I want to use it more. Okay, I'll try again, and if it doesn't work, I'll revert it.